schemas.js

```diff
@@ -24,12 +24,38 @@ const extension = (joi) => ({
 
 const Joi = BaseJoi.extend(extension);
 
-module.exports.campgroundSchema = Joi.object({
-  campground: Joi.object({
-    title: Joi.string().required().escapeHTML(),
-    price: Joi.number().required().min(0),
-    location: Joi.string().required().escapeHTML(),
-    description: Joi.string().required().escapeHTML(),
+module.exports.sendPlayDataSchema = Joi.object({
+  difficulty: Joi.number().max(5).required(),
+  sessionLength: Joi.number().required(),
+});
+module.exports.updatePlayDataSchema = Joi.object({
+  guess: Joi.string().required(),
+  elemId: Joi.string().required(),
+  additionalData: Joi.object({
+    sessionId: Joi.string().uuid().required(),
+    pageNum: Joi.number().integer().required(),
+    action: Joi.string().valid("next", "back").required(),
   }).required(),
-  deleteImages: Joi.array(),
+});
+module.exports.registrationSchema = Joi.object({
+  username: Joi.string().alphanum().min(7).max(30).required(),
```

Comment:
It's good that you set `min` and `max` values although I would say that you should also set these properties to the HTML elements as well in the sign up form. Examples how to do that are here: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/text. The reason why I'm saying this is because for the user it's probably not gonna be clear why their usernames and passwords are invalid.

Reply:
I absolutely agree with that. Main reason why I haven't done it, I wasn't quite sure if the stuff I'm doing with schemas here is all correct, and wanted to have a discussion on whether we should add a separate task for adding some sort of user feedback related to newly added schema validation was necessary.